From 594c6b276347773edb4bd66fcfb5bb230fad77aa Mon Sep 17 00:00:00 2001
From: Karthik M <quic_karm@quicinc.com>
Date: Tue, 26 Sep 2023 18:38:23 +0530
Subject: [PATCH] netifd-wireless.sh: Add AKM25 support

Add support for AKM25 with GCMP as wpa_cipher and
FT-SAE-EXT-KEY as key management.

Uses SAE for authentication and based on KCK length,
hashing algorithm is detemined.

ECC length   FCC length   HASH algo
p<=256       p<=2048      SHA-256
256<p<=384   2048<p<=3072 SHA-384
384<p        3072<p       SHA-512

Signed-off-by: Karthik M <quic_karm@quicinc.com>
---
 scripts/netifd-wireless.sh | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/scripts/netifd-wireless.sh b/scripts/netifd-wireless.sh
index 25b089b..146f084 100644
--- a/scripts/netifd-wireless.sh
+++ b/scripts/netifd-wireless.sh
@@ -225,7 +225,7 @@ wireless_vif_parse_encryption() {
 		*aes|*ccmp| psk2 | wpa2 | sae | owe) wpa_cipher="CCMP";;
 		*tkip | wpa | psk ) wpa_cipher="TKIP";;
 		*gcmp256) wpa_cipher="GCMP-256";;
-		*gcmp | sae-ext-key) wpa_cipher="GCMP";;
+		*gcmp | *sae-ext-key) wpa_cipher="GCMP";;
 		wpa3-192*) wpa_cipher="GCMP-256";;
 	esac
 
@@ -235,7 +235,7 @@ wireless_vif_parse_encryption() {
 	# wpa2/tkip+aes     => WPA2 RADIUS, CCMP+TKIP
 
 	case "$encryption" in
-		wpa2*|wpa3*|*psk2*|psk3*|sae*|owe*)
+		wpa2*|wpa3*|*psk2*|psk3*|*sae*|owe*)
 			wpa=2
 		;;
 		wpa*mixed*|*psk*mixed*)
@@ -306,6 +306,9 @@ wireless_vif_parse_encryption() {
 		sae-mixed)
 			auth_type=sae-mixed
 		;;
+		ft-sae-ext-key)
+			auth_type=ft-sae-ext-key
+		;;
 	esac
 
 	case "$encryption" in
-- 
2.17.1

